editor/plugins/canvas_item_editor_plugin.cpp

```diff
@@ -963,8 +963,33 @@ void CanvasItemEditor::_restore_canvas_item_state(List<CanvasItem *> p_canvas_it
 }
 
 void CanvasItemEditor::_commit_canvas_item_state(List<CanvasItem *> p_canvas_items, String action_name, bool commit_bones) {
-	undo_redo->create_action(action_name);
+	List<CanvasItem *> modified_canvas_items;
 	for (List<CanvasItem *>::Element *E = p_canvas_items.front(); E; E = E->next()) {
+		CanvasItem *canvas_item = E->get();
+		Dictionary old_state = editor_selection->get_node_editor_data<CanvasItemEditorSelectedItem>(canvas_item)->undo_state;
+		Dictionary new_state = canvas_item->_edit_get_state();
+
+		// Can't compare Dictionaries with ==.
+		bool different = false;
+		Array keys = new_state.keys();
+		for (int i = 0; i < keys.size(); i++) {
+			if (old_state.get(keys[i], Variant()) != new_state[keys[i]]) {
+				different = true;
+				break;
+			}
+		}
```

Comment:
Might be faster and simpler to compare `.hash()`?

Reply:
```gdscript
var dict1 = {
	"a": 0,
}

var dict2 = {
	"a": 0,
}

func _ready():
	print(dict1 == dict2)  # False
	print(dict1.hash() == dict2.hash())  # True
```

Reply:
Right, I forgot about this...